src/main/java/org/prebid/server/analytics/reporter/greenbids/model/HttpUtil.java

```diff
@@ -0,0 +1,65 @@
+package org.prebid.server.analytics.reporter.greenbids.model;
+
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.prebid.server.deals.PlannerService;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Scanner;
+
+public class HttpUtil {
+
+    private HttpUtil() {
+        throw new UnsupportedOperationException("Utility class and cannot be instantiated");
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(PlannerService.class);
+
+    public static void sendJson(String json, String endpointUrl) {
```

Comment:
can you use java native HttpClient for this purpose? Because the following looks like an overkill

Reply:
Have applied the [HttpClient.post](https://github.com/EvgeniiMunin/prebid-server-java/blob/4ab3caea2316a5ff04d0961be9a4e3c4896d517b/src/main/java/org/prebid/server/vertx/httpclient/HttpClient.java#L41) as follows

```java
final MultiMap headers = MultiMap.caseInsensitiveMultiMap()
           .add(HttpUtil.ACCEPT_HEADER, HttpHeaderValues.APPLICATION_JSON);


Future<HttpClientResponse> responseFuture = httpClient.post(
             greenbidsAnalyticsProperties.getAnalyticsServer(),
             headers,
             commonMessageJson,
             greenbidsAnalyticsProperties.getTimeoutMs());

 responseFuture
            .onSuccess(response -> logger.info("Analytics Server response body: " + response.getStatusCode() + "\n" + response.getHeaders() + "\n" + response.getBody()))
            .onFailure(error -> logger.error("Can't send payload to Analytics Server: " + error));
```

but get the status code 500 with the following response body
```
<!doctype html>
<html lang=en>
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>
```

At the same time when sending the same request on custom HttpUtil.sendJson the status code is 202 and the request is validated by analytics server

Reply:
Could you please help me to figure out why `httpClient.post` request returns 500 status code?

Reply:
is it the error from Analytics Server? Why is it in the html format instead of json? Maybe the `content-type` is missing?

Reply:
Yes, thanks! The reason was actually to set the content type header

```java
final MultiMap headers = MultiMap.caseInsensitiveMultiMap()
     .add(HttpUtil.ACCEPT_HEADER, HttpHeaderValues.APPLICATION_JSON)
     .add(HttpUtil.CONTENT_TYPE_HEADER, HttpHeaderValues.APPLICATION_JSON);
```

After the fix get the 202 status code from analytics server

```
Analytics Server response body: 202
server=Werkzeug/3.0.3 Python/3.11.9
date=Tue, 21 May 2024 14:24:51 GMT
content-type=application/json
Content-Length=140
access-control-allow-origin=*
via=1.1 google
Alt-Svc=h3=":443"; ma=2592000,h3-29=":443"; ma=2592000

{"type": "greenbids/publishers/exceptions/TestRequest", "title": "TestRequest", "status": 202, "detail": "Request sent with the test pbuid"}
```